src/core/async.hpp

```diff
@@ -20,79 +112,100 @@ struct SharedThreadPoolConcurrency {
   }
 };
 #endif
+
 extern Shared<boost::asio::thread_pool, SharedThreadPoolConcurrency> SharedThreadPool;
 
 template <typename FUNC, typename CANCELLATION>
 inline SHVar awaitne(SHContext *context, FUNC &&func, CANCELLATION &&cancel) noexcept {
-  // SHLOG_TRACE("awaitne starting");
-  // DEFER(SHLOG_TRACE("awaitne ending"));
 #if defined(__EMSCRIPTEN__) && !defined(__EMSCRIPTEN_PTHREADS__)
   return func();
 #else
-  SHVar res{};
-  std::packaged_task<SHVar()> task(std::forward<FUNC>(func));
-  auto future = task.get_future();
+  struct BlockingCall : TidePool::Work {
+    BlockingCall(FUNC &&func) : func(std::move(func)), exp(), res(), complete(false) {}
+
+    virtual ~BlockingCall() {}
```

Comment:
Doesn't need virtual destructor, only if you were to destroy it from it's base class, but then you'd need it on Work instead

Reply:
I had a warning, that's why I added it in test_runtime which is similar.
Doesn't hurt anyway does it?

Reply:
no warning in this header though seems indeed

Reply:
err at least on win, will remove for now we see